File: flight/src/main/java/com/airline/flight/dto/FlightDto.java
package com.airline.flight.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FlightDto {
    private String flightId;
    private String airline;
    private String departure;
    private String arrival;
    private LocalDateTime departureTime;
    private LocalDateTime arrivalTime;
    private BigDecimal price;
    private Integer availableSeats;
    private String aircraft;
}

File: reservation/src/main/kotlin/com/airline/reservation/dto/external/FlightDto.kt
package com.airline.reservation.dto.external

import java.math.BigDecimal
import java.time.LocalDateTime

data class FlightDto(
    val flightId: String,
    val airline: String,
    val departure: String,
    val arrival: String,
    val departureTime: LocalDateTime,
    val arrivalTime: LocalDateTime,
    val price: BigDecimal,
    val availableSeats: Int,
    val aircraft: String
)

data class AvailabilityRequest(
    val flightId: String? = null,
    val requestedSeats: Int
)

data class AvailabilityResponse(
    val available: Boolean,
    val flightId: String,
    val availableSeats: Int,
    val message: String
)